moveit_core/kinematic_constraints/src/utils.cpp

```diff
@@ -165,6 +160,23 @@ moveit_msgs::msg::Constraints constructGoalConstraints(const moveit::core::Robot
   return goal;
 }
 
+bool updateConstraints(moveit_msgs::msg::Constraints& constraints, const moveit::core::RobotState& state,
+                       const moveit::core::JointModelGroup* jmg)
+{
+  std::vector<double> vals;
+  state.copyJointGroupPositions(jmg, vals);
+  if (constraints.joint_constraints.size() != vals.size())
```

Comment:
If you have to resize the joint constraints, you are using a completely different JMG which would require also initializing `joint_name`, `weight` and tolerances. If you really just want to update the position values, I would rule out this case by only iterating the constraints and updating the values of those joints that are part of the group.

I would even say that it should explicitly only update the constraints values that are already present without adding or removing new ones. Otherwise, a new constraint should be constructed with all the required member values.

Reply:
That makes a lot of sense. Done.